metaserver/src/main/scala/scala/meta/languageserver/compiler/CompletionProvider.scala

```diff
@@ -32,6 +32,27 @@ object CompletionProvider extends LazyLogging {
       else CompletionItemKind.Value
     }
 
+    def computeRelevance(
+        sym: Symbol,
+        viaView: Symbol,
+        inherited: Boolean
+    ): Int = {
+      var relevance = 0
+      if (sym.isLocalToBlock) relevance += 10
+      if (sym.hasGetter) relevance += 5
+      if (!inherited) relevance += 10
+      if (viaView == NoSymbol) relevance += 20
+      if (!sym.hasPackageFlag) relevance += 30
+      if (sym.isCaseAccessor) relevance += 10
+      if (sym.isPublic) relevance += 10
+      if (sym.owner != definitions.AnyClass && sym.owner != definitions.AnyRefClass && sym.owner != definitions.ObjectClass)
+        relevance += 40
+      if (sym.owner != null && sym.owner.isCaseClass && sym.nameString == "copy") {
```

Comment:
here I'm penalizying `copy` on case classes since I want it after user-defined methods/vals.
Any other similar cases that you guys can think of?
Does somebody know a way to check whether a symbol is synthesized by the compiler?

Reply:
`.isSynthetic` may do the trick, I think that will also catch synthetic productArity,productIterator,productPrefix,productElement methods.

Reply:
Could you add this as a comment? Also it's not obvious to me at the first sight why the increment is different for some cases. I think adding some comments for special cases would help a lot in reading this code later.

Reply:
Thank you @olafurpg, `.isSynthetic` seems to be working :)

@laughedelic I'll add some comments, also specifying that I've essentially ported https://github.com/scala-ide/scala-ide/blob/a17ace0ee1be1875b8992664069d8ad26162eeee/org.scala-ide.sdt.core/src/org/scalaide/core/completion/ProposalRelevanceCalculator.scala plus some tweaks. I've respected the weights of the original implementation, but I don't think it's an exact science.

The only weight that I blindly ported without really understanding why is `hasPackageFlag`. I have a vague understanding of what it means, but maybe @olafurpg can shed some light?